HealthcareSystem/GraphicalEditor/MainWindow.xaml.cs

```diff
@@ -127,48 +133,28 @@ private MapObject findMapObjectWithMinimumArea(List<MapObject> MapObjects)
             {
                 return null;
             }
-            
+
         }
 
 
         private void Canvas_MouseDown(object sender, MouseButtonEventArgs e)
         {
-            _mouseCurrentPosition = e.GetPosition(this.Canvas);
-
-            MapObject selectedMapObject = findSelectedMapObject();
+            MapObject selectedMapObject = FindSelectedMapObject(e.GetPosition(this.Canvas));
 
-            if(selectedMapObject != null)
+            if (selectedMapObject != null)
             {
                 TestText.Text = selectedMapObject.MapObjectEntity.MapObjectType.ObjectTypeFullName;
             }
             else
             {
                 TestText.Text = "";
             }
-            
-        }
-
-        private void Canvas_MouseUp(object sender, MouseButtonEventArgs e)
-        {
 
         }
 
-
-        private bool isMouseXCoordinateInsideRectangle(double mouseXCoordinate, Rectangle objectRectangle)
-        {
-            if (mouseXCoordinate > Canvas.GetLeft(objectRectangle) && mouseXCoordinate < Canvas.GetLeft(objectRectangle) + objectRectangle.Width)
-            {
-                return true;
-            }
-            else
-            {
-                return false;
-            }
-        }
-
-        private bool isMouseYCoordinateInsideRectangle(double mouseYCoordinate, Rectangle objectRectangle)
+        private bool IsMouseCursorInsideRectangle(Point cursorPosition, Rectangle objectRectangle)
         {
-            if ((mouseYCoordinate > Canvas.GetTop(objectRectangle)) && (mouseYCoordinate < Canvas.GetTop(objectRectangle) + objectRectangle.Height))
+            if ((cursorPosition.X > Canvas.GetLeft(objectRectangle) && cursorPosition.X < Canvas.GetLeft(objectRectangle) + objectRectangle.Width) && ((cursorPosition.Y > Canvas.GetTop(objectRectangle) && cursorPosition.Y < Canvas.GetTop(objectRectangle) + objectRectangle.Height)))
```

Comment:
Maybe split this into multiple ifs for readability? Something like:
if (cursorPosition.X < Canvas.getLeft(objectRectangle) return false;
if (cursonPositionX > (Canvas.getLeft(objectRectangle) + objectRectangle.Width)) return false;
...
That's not really ideal either though, so I'm not sure what the best way to write this is.

Reply:
Okay, I will take that into consideration, although I prefer it the way I did it, but I agree that it is very questionable what is the best way to refactor it. Thanks for feedback and advices! 😄